TASVideos.Common/HtmlWriter.cs

```diff
@@ -4,6 +4,142 @@ namespace TASVideos.Common;
 
 public partial class HtmlWriter(TextWriter w)
 {
+	private struct HtmlClassList() // : ISet<string>, IReadOnlySet<string>
+	{
+		private List<string>? _list = null;
+
+		private string? _single = null;
+
+/*
+		public readonly int Count
+			=> throw new NotImplementedException();
+
+		public readonly bool IsReadOnly
+			=> false;
+*/
+
+		public /*bool*/void Add(string item)
+		{
+			if (_single is not null)
+			{
+				_list = new() { _single, item };
+				_single = null;
+			}
+			else if (_list is not null)
+			{
+				_list.Add(item);
+			}
+			else
+			{
+				_single = item;
+			}
+		}
+
+/*
+		void ICollection<string>.Add(string item)
+			=> _ = Add(item);
+
+		public void Clear()
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly bool Contains(string item)
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly void CopyTo(string[] array, int arrayIndex)
+		{
+			throw new NotImplementedException();
+		}
+
+		public void ExceptWith(IEnumerable<string> other)
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly IEnumerator<string> GetEnumerator()
+		{
+			throw new NotImplementedException();
+		}
+
+		readonly IEnumerator IEnumerable.GetEnumerator()
+			=> GetEnumerator();
+
+		public void IntersectWith(IEnumerable<string> other)
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly bool IsProperSubsetOf(IEnumerable<string> other)
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly bool IsProperSupersetOf(IEnumerable<string> other)
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly bool IsSubsetOf(IEnumerable<string> other)
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly bool IsSupersetOf(IEnumerable<string> other)
+		{
+			throw new NotImplementedException();
+		}
+
+		public readonly bool Overlaps(IEnumerable<string> other)
+		{
+			throw new NotImplementedException();
+		}
+
+		public bool Remove(string item)
+		{
+			throw new NotImplementedException();
+		}
+*/
+
+		public readonly string Serialise()
+		{
+			HashSet<string> classes;
+			if (!string.IsNullOrEmpty(_single))
+			{
+				classes = _single.Split(' ', StringSplitOptions.RemoveEmptyEntries).ToHashSet();
+			}
+			else if (_list is not null)
+			{
+				classes = _list.SelectMany(static s => s.Split(' ', StringSplitOptions.RemoveEmptyEntries)).ToHashSet();
+			}
+			else
+			{
+				return string.Empty;
+			}
+
+			return classes.Count is 0 ? string.Empty : $" class=\"{string.Join(' ', classes)}\"";
+		}
+
+/*
+		public readonly bool SetEquals(IEnumerable<string> other)
```

Comment:
Didn't end up using the `ISet` impl., and I don't see that being necessary in the future. Will remove.